code/modules/reagents/chemistry/machinery/chem_dispenser.dm

```diff
@@ -276,10 +343,76 @@
 	max_energy = 100
 	desc = "A technological marvel, supposedly able to mix just the mixture you'd like to drink the moment you ask for one."
 	dispensable_reagents = list("ice", "cream", "cider", "beer", "kahlua", "whiskey", "wine", "vodka", "gin", "rum", "tequila", "vermouth", "cognac", "ale", "mead", "synthanol")
+	/var/list/special_reagents3 = list(list("iced_beer"),
+								list("irishcream", "manhattan",),
+								list("antihol", "synthignon", "bravebull"))
 	hack_message = "You disable the 'nanotrasen-are-cheap-bastards' lock, enabling hidden and very expensive boozes."
 	unhack_message = "You re-enable the 'nanotrasen-are-cheap-bastards' lock, disabling hidden and very expensive boozes."
 	hacked_reagents = list("goldschlager", "patron", "absinthe", "ethanol", "nothing", "sake")
 
+/obj/machinery/chem_dispenser/beer/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/beer(null)
+	component_parts += new /obj/item/stock_parts/matter_bin(null)
+	component_parts += new /obj/item/stock_parts/matter_bin(null)
+	component_parts += new /obj/item/stock_parts/manipulator(null)
+	component_parts += new /obj/item/stock_parts/capacitor(null)
+	component_parts += new /obj/item/stock_parts/console_screen(null)
+	component_parts += new /obj/item/stock_parts/cell/super(null)
+	RefreshParts()
+
+/obj/machinery/chem_dispenser/beer/upgraded/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/circuitboard/beer(null)
+	component_parts += new /obj/item/stock_parts/matter_bin/super(null)
+	component_parts += new /obj/item/stock_parts/matter_bin/super(null)
+	component_parts += new /obj/item/stock_parts/manipulator/pico(null)
+	component_parts += new /obj/item/stock_parts/capacitor/super(null)
+	component_parts += new /obj/item/stock_parts/console_screen(null)
+	component_parts += new /obj/item/stock_parts/cell/hyper(null)
+	RefreshParts()
+
+/obj/machinery/chem_dispenser/beer/RefreshParts()
+	var/i
+	for(var/obj/item/stock_parts/manipulator/M in component_parts)
+		for(i=1, i<=M.rating, i++)
```

Comment:
follow the style guide.

Reply:
Same as above.

Reply:
Definitely not sure at all how to change or edit this. Again, for me, the style guide isn't really clear on a lot of things.

Reply:
`for(var/i in 1 to M.rating)`

Then remove the bit above the `for` proc where you declare `i`

Reply:
Ooh...even though it took someone else to point out to me what you meant, thanks, xD